server/src/test/java/org/elasticsearch/cluster/routing/allocation/ShardWriteLoadDistributionMetricsTests.java

```diff
@@ -0,0 +1,463 @@
+                        ShardWriteLoadDistributionMetrics.shardWriteLoadDistributionMetricName(percentile)
+                    ),
+                empty()
+            );
+        }
+        assertThat(
+            testInfrastructure.meterRegistry.getRecorder()
+                .getMeasurements(
+                    InstrumentType.DOUBLE_GAUGE,
+                    ShardWriteLoadDistributionMetrics.WRITE_LOAD_PRIORITISATION_THRESHOLD_METRIC_NAME
+                ),
+            empty()
+        );
+        assertThat(
+            testInfrastructure.meterRegistry.getRecorder()
+                .getMeasurements(
+                    InstrumentType.LONG_GAUGE,
+                    ShardWriteLoadDistributionMetrics.WRITE_LOAD_PRIORITISATION_THRESHOLD_PERCENTILE_RANK_METRIC_NAME
+                ),
+            empty()
+        );
+    }
+
+    public TestInfrastructure createTestInfrastructure() {
+        final RecordingMeterRegistry meterRegistry = new RecordingMeterRegistry();
+        final ClusterService clusterService = mock(ClusterService.class);
+        final var settings = Settings.builder()
+            .put(ShardWriteLoadDistributionMetrics.SHARD_WRITE_LOAD_METRICS_ENABLED_SETTING.getKey(), true)
+            .put(DiscoveryNode.STATELESS_ENABLED_SETTING_NAME, true)
+            .build();
+        when(clusterService.getClusterSettings()).thenReturn(ClusterSettings.createBuiltInClusterSettings(settings));
+        when(clusterService.getSettings()).thenReturn(settings);
+        when(clusterService.lifecycleState()).thenReturn(Lifecycle.State.STARTED);
+        final var indexName = randomIdentifier();
+        final var clusterState = balanceShardsByCount(
+            ClusterStateCreationUtils.buildServerlessRoleNodes(indexName, 200, 2, randomIntBetween(1, 2), randomIntBetween(1, 2))
+        );
+        when(clusterService.state()).thenReturn(clusterState);
+        final int numberOfSignificantDigits = randomIntBetween(2, 3);
+        final ShardWriteLoadDistributionMetrics shardWriteLoadDistributionMetrics = new ShardWriteLoadDistributionMetrics(
+            meterRegistry,
+            clusterService,
+            numberOfSignificantDigits,
+            0,
+            50,
+            90,
+            100
+        );
+        final double maxp50 = randomDoubleBetween(0, 10, true);
+        final double maxp90 = randomDoubleBetween(maxp50, 30, true);
+        final double maxp100 = randomDoubleBetween(maxp90, 50, true);
+
+        final var clusterInfo = ClusterInfo.builder().shardWriteLoads(randomWriteLoads(clusterState, maxp50, maxp90, maxp100)).build();
+        return new TestInfrastructure(
+            clusterService,
+            meterRegistry,
+            clusterInfo,
+            shardWriteLoadDistributionMetrics,
+            numberOfSignificantDigits,
+            maxp50,
+            maxp90,
+            maxp100
+        );
+    }
+
+    public record TestInfrastructure(
+        ClusterService clusterService,
```

Comment:
```suggestion
        ClusterService mockClusterService,
```
?

Reply:
Renamed in 25441c8bd09301372d69969df05e812c8d745445